feat(auth): add OAuth Access Token support to sign in with providers (#9593)
firebase/flutterfire: docs/auth/federated-auth.md

}
```

# Using the OAuth access token

> By using an AuthProvider, you can retrieve the access token associated with the provider
> and make request to the provider.

Reviewer: ```suggestion
By using an AuthProvider, you can retrieve the access token associated with the provider
by making the following request.
```

```dart
final appleProvider = AppleAuthProvider();

final user = await FirebaseAuth.instance.signInWithProvider(appleProvider);
final accessToken = user.credential?.accessToken;

> // You can send requests with `accessToken`

Reviewer: ```suggestion
// You can send requests with the `accessToken`
```
